### PR TITLE
Prep mc for exp release

### PR DESCRIPTION
**Pull request by Captain-Coder**
Continuation of PR #2394.

### CODE REVIEW

**Comment by Captain-Coder** [comment]
retest this please, now some channel tests are failing during tear down ("No channel named de"?). But atleast the tunnel tests work now.

**Comment by devos50** [comment]
@Captain-Coder that's because our search tests are still dependent on external Dispersy node. It does not happen frequently and I hope it also gets better with #2454.

**Comment by Captain-Coder** [comment]
@devos50 Any idea what happened this time? I get usefull test failures like [this](https://jenkins.tribler.org/job/GH_Tribler_PR_tests_linux/2620/testReport/junit/Tribler.Test.Core.test_sqlitecachedbhandler_votecasts/TestVotecastDBHandler/test_get_my_votes/) though i suspect [this](https://jenkins.tribler.org/job/GH_Tribler_PR_tests_linux/2620/testReport/junit/Tribler.Test.test_my_channel/TestMyChannel/test_add_torrents_playlists/) to be the root cause. My original bugs are solved, and it passed consistently except for those on the old PR, so i'm very much inclined to just put this one on ready.

**Comment by devos50** [comment]
@Captain-Coder please refer to the test logs, see https://jenkins.tribler.org/job/GH_Tribler_PR_tests_linux/2620/artifact/output/test_runners_output/00001.err/*view*/ for instance.

There was a dirty reactor. This one is known (there is an issue with this error somewhere) but it's in Dispersy.

There's another error here: https://jenkins.tribler.org/job/GH_Tribler_PR_tests_linux/2620/artifact/output/test_runners_output/00003.err/*view*/.

**Comment by whirm** [comment]
retest this please

**Comment by Captain-Coder** [comment]
Looks good to me

**Comment by whirm** [comment]
@devos50 did you give the OK on this one?

File: Tribler/Test/Community/Multichain/test_multichain_database.py [hunk 1]
```diff
@@ -204,3 +204,27 @@ def test_get_insert_time(self):
         self.assertEquals(time_difference.days, 0)
         self.assertLess(time_difference.seconds, 10,
                         "Difference in stored and retrieved time is too large.")
+
+    def test_database_upgrade(self):
+        self.db.executescript(u"UPDATE option SET value = '1' WHERE key = 'database_version';")
+        self.db.close(commit=True)
+        self.db = None
```
Comment (devos50) [inline Tribler/Test/Community/Multichain/test_multichain_database.py]: I guess these sequences of closing the database can be refactored into a separate method?

File: Tribler/community/multichain/database.py [hunk 2]
```diff
@@ -286,7 +286,11 @@ def check_database(self, database_version):
         assert int(database_version) >= 0
         database_version = int(database_version)
 
-        if database_version < 1:
+        if database_version < LATEST_DB_VERSION:
+            # Remove all previous data, since we have only been testing so far, and previous blocks might not be
+            # reliable. In the future, we should implement an actual upgrade procedure
+            self.executescript(u"DROP TABLE IF EXISTS multi_chain")
```
Comment (devos50) [inline Tribler/community/multichain/database.py]: Can this be done in one method call? Something like (with each statement on a separate line):

```
self.executescript(u"DROP TABLE IF EXISTS multi_chain; DROP TABLE IF EXISTS option;");
```

**Comment by devos50** [comment]
@whirm @Captain-Coder I wasn't notified of reviewing this PR. I gave some feedback ;)

**Comment by Captain-Coder** [comment]
retest this please, random dirty reactor on a single mac test case (Tribler.Test.Core.Modules.RestApi.Channels.test_channels_torrents_endpoint.TestModifyChannelTorrentEndpoint.test_add_torrent_from_url_to_channel_with_description)

**Comment by Captain-Coder** [comment]
retest this please.
- on linux, test_i2i_download got "Couldn't listen on 127.0.0.1:57891: [Errno 98] Address already in use."
- on win64, "test rss source" because of a dirty reactor, "testing existing channel as a source" because of test timeout, "Test the restriction of max_torrents in a source" because of test timeout, "API returns error 400 if an unknown state is passed when modifying a download" failes because "exceptions.TypeError: argument of type 'NoneType' is not iterable", anything after that test fails with "Exception: Test raised 1 unhandled Twisted exceptions:" even though no exceptions are printed.
- on mac, "API returns the right JSON data if a channel is created" runs into "Connection was refused by other side: 61: Connection refused", anythng after that is "Exception: Test raised 2 unhandled Twisted exceptions:" even though no exceptions are printed.

**Comment by synctext** [comment]
retest this please

**Comment by Captain-Coder** [comment]
But it was already green, and without random failures this time... :P

**Pull request merged**